File: src/test/java/utils/RandomAccessFileCacheTest.java
package utils;

import de.esymetric.jerusalem.utils.RandomAccessFileCache;
import org.junit.Test;

import java.io.RandomAccessFile;

import static org.junit.Assert.assertEquals;

public class RandomAccessFileCacheTest {
    @Test
    public void testRandomAccessFileCache() throws Exception {
        RandomAccessFileCache rafCache = new RandomAccessFileCache();

        for (int i = 0; i <= 30; i++) {
            RandomAccessFile raf = rafCache.getRandomAccessFile(
                    "testData/rafCache" + i + ".data", false);
            for (int j = 0; j < 200000; j++) {
                int p = (int)(Math.random() * 20000);
                raf.seek(p);
                raf.writeInt(j);
            }
            raf.seek(999);
            raf.writeInt(333);
        }

        for (int i = 10; i > 0; i--) {
            RandomAccessFile raf = rafCache.getRandomAccessFile(
                    "testData/rafCache" + i + ".data", false);
            raf.seek(999);
            assertEquals(333, raf.readInt());
        }

        rafCache.close();
    }

}

File: src/test/java/utils/RandomAccessFileCacheTest.kt
package utils

import de.esymetric.jerusalem.utils.RandomAccessFileCache
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

class RandomAccessFileCacheTest {
    @Test
    @Throws(Exception::class)
    fun testRandomAccessFileCache() {
        val rafCache = RandomAccessFileCache()
        for (i in 0..30) {
            val raf = rafCache.getRandomAccessFile(
                "testData/rafCache$i.data", false
            )
            for (j in 0..199999) {
                val p = (Math.random() * 20000).toInt()
                raf!!.seek(p.toLong())
                raf.writeInt(j)
            }
            raf!!.seek(999)
            raf.writeInt(333)
        }
        for (i in 10 downTo 1) {
            val raf = rafCache.getRandomAccessFile(
                "testData/rafCache$i.data", false
            )
            raf!!.seek(999)
            assertEquals(333, raf.readInt().toLong())
        }
        rafCache.close()
    }
}